cabal-install/src/Distribution/Client/ProjectPlanning.hs

```diff
@@ -2224,14 +2225,28 @@ elaborateInstallPlan
 
             elabPkgDescriptionOverride = descOverride
 
-            elabVanillaLib = perPkgOptionFlag pkgid True packageConfigVanillaLib -- TODO: [required feature]: also needs to be handled recursively
-            elabSharedLib = pkgid `Set.member` pkgsUseSharedLibrary
-            elabStaticLib = perPkgOptionFlag pkgid False packageConfigStaticLib
-            elabDynExe = perPkgOptionFlag pkgid False packageConfigDynExe
-            elabFullyStaticExe = perPkgOptionFlag pkgid False packageConfigFullyStaticExe
-            elabGHCiLib = perPkgOptionFlag pkgid False packageConfigGHCiLib -- TODO: [required feature] needs to default to enabled on windows still
-            elabProfExe = perPkgOptionFlag pkgid False packageConfigProf
-            elabProfLib = pkgid `Set.member` pkgsUseProfilingLibrary
+            elabBuildOptions =
+              LBC.BuildOptions
+                { withVanillaLib = perPkgOptionFlag pkgid True packageConfigVanillaLib -- TODO: [required feature]: also needs to be handled recursively
+                , withSharedLib = pkgid `Set.member` pkgsUseSharedLibrary
+                , withStaticLib = perPkgOptionFlag pkgid False packageConfigStaticLib
+                , withDynExe = perPkgOptionFlag pkgid False packageConfigDynExe
+                , withFullyStaticExe = perPkgOptionFlag pkgid False packageConfigFullyStaticExe
+                , withGHCiLib = perPkgOptionFlag pkgid False packageConfigGHCiLib -- TODO: [required feature] needs to default to enabled on windows still
+                , withProfExe = perPkgOptionFlag pkgid False packageConfigProf
+                , withProfLib = pkgid `Set.member` pkgsUseProfilingLibrary
+                , exeCoverage = perPkgOptionFlag pkgid False packageConfigCoverage
+                , libCoverage = perPkgOptionFlag pkgid False packageConfigCoverage
+                , withOptimization = perPkgOptionFlag pkgid NormalOptimisation packageConfigOptimization
+                , splitObjs = perPkgOptionFlag pkgid False packageConfigSplitObjs
+                , splitSections = perPkgOptionFlag pkgid False packageConfigSplitSections
+                , stripLibs = perPkgOptionFlag pkgid False packageConfigStripLibs
+                , stripExes = perPkgOptionFlag pkgid False packageConfigStripExes
+                , withDebugInfo = perPkgOptionFlag pkgid NoDebugInfo packageConfigDebugInfo
+                , relocatable = perPkgOptionFlag pkgid False packageConfigRelocatable
+                , withProfLibDetail = elabProfExeDetail
+                , withProfExeDetail = elabProfLibDetail
+                }
```

Comment:
This is great, ElaboratedConfiguredPackage is mixed bag of data, each relevant in different parts of the build. It is great to see it regrouped.